### PR TITLE
[ENH] Validation split fix issue 7644

### LINKED ISSUE #7644: [BUG] TinyTimeMixer should not create validation set if validation_split is None
**Describe the bug**

`TinyTimeMixerForecaster` always creates a validation set even if `validation_split is None`, in which case the `temporal_train_test_split` defaults the `validation_split` to `0.25` 

Following fixes should be made:
- create `y_test` only if `validation_split is not None`
- rename data split: `test` to `eval`

**To Reproduce**

```python
from sktime.forecasting.ttm import TinyTimeMixerForecaster
from sktime.datasets import load_airline
y = load_airline()

forecaster = TinyTimeMixerForecaster(
	config={"context_length": 10},
	training_args={
		"output_dir": "test_output",
		"report_to": "none",
	},
	validation_split=None, # defaults validation_split to 0.25
)
forecaster.fit(y, fh=[1, 2, 3])
y_pred = forecaster.predict()
```

**Expected behavior**

when `validation_split is None`, `test` should not be created and not passed to the `Trainer`

**Versions**
<details>

```
System:
    python: 3.12.5 | packaged by Anaconda, Inc. | (main, Sep 12 2024, 18:27:27) [GCC 11.2.0]
executable: /home/geetu/miniconda3/envs/ai/bin/python
   machine: Linux-6.8.0-51-generic-x86_64-with-glibc2.39

Python dependencies:
          pip: 24.2
       sktime: 0.34.0
      sklearn: 1.5.2
       skbase: 0.8.3
        numpy: 1.26.4
        scipy: 1.14.1
       pandas: 2.2.3
   matplotlib: 3.9.2
       joblib: 1.4.2
        numba: None
  statsmodels: None
     pmdarima: None
statsforecast: None
      tsfresh: None
      tslearn: None
        torch: 2.4.1
   tensorflow: 2.18.0
```

</details>

### PR DESCRIPTION
**Pull request by RHYTHM2405**
Description:

This update (Fixes #7644) modifies the code to ensure that the y_test dataset is only created when validation_split is not None. If validation_split is None, no y_test dataset is created, and it is not passed to the Trainer. This change optimizes resource usage and eliminates unnecessary processing when validation data is not required.

Key Changes:

Added a conditional check to create y_test only when validation_split is provided.
Ensured that y_test is excluded from being passed to the Trainer when validation_split is None.

### CODE REVIEW

**Review by yarnabrina — CHANGES_REQUESTED** [review]
Thank you for your contribution.

I think it'll be less duplication if you create a dictionary of argument names and values to pass, starting with common ones and adding evaluation dataset depending on the check. Then you can make a single call with dictionary unpacking.

Also, if you haven't already done, please ensure to follow contribution guidelines and use pre-commit during development.

**Review by geetu040 — CHANGES_REQUESTED** [review]
@RHYTHM2405 you have made changes in the wrong file, please change only the [TinyTimeMixerForecaster](https://github.com/sktime/sktime/blob/main/sktime/forecasting/ttm.py).

`HFTransformersForecaster` already handles this, you can take the idea from there

**Comment by RHYTHM2405** [comment]
@geetu040 Sure, I've made changes in ttm.py where I have renamed test to eval . Also I have a functionality which checks whether y_test needs to be created or not . @yarnabrina I have also added a dictionary which includes `eval_dataset` only if test is `not_none`

File: sktime/forecasting/ttm.py [hunk 1]
```diff
@@ -365,15 +365,21 @@ def _fit(self, y, X, fh):
...
             y=y_train,
             context_length=config.context_length,
             prediction_length=config.prediction_length,
         )
         test = PyTorchDataset(
-            y=y_test,
+            y=y_eval,
             context_length=config.context_length,
             prediction_length=config.prediction_length,
         )
```
Comment (geetu040) [inline sktime/forecasting/ttm.py:385]: - create `test` only if `validation_split is not None`
- rename `test` to `eval`

**Review by geetu040 — CHANGES_REQUESTED** [review]
Thanks! a few changes requested

**Review by geetu040 — APPROVED** [review]
LGTM!
@RHYTHM2405 Thank you for the fix.

**Review by yarnabrina — APPROVED** [review]
